src/lib/symlink-manager.ts

```diff
@@ -0,0 +1,263 @@
+	}
+
+	/**
+	 * Add a symlink to the PHP runtime by mounting the target path.
+	 * @param filename
+	 */
+	private async addSymlink( filename: string ) {
+		const fullPath = path.join( this.projectPath, filename );
+		const target = await fs.realpath( fullPath );
+
+		if ( ! this.php.requestHandler ) {
+			throw new Error( 'Request handler is not set' );
+		}
+
+		const vfsPath = path.posix.join( this.php.requestHandler.documentRoot, filename );
+
+		// Double check to ensure the symlink exists
+		if ( ! this.php.fileExists( vfsPath ) ) {
+			return;
+		}
+
+		// Get the realpath of the symlink within the PHP runtime.
+		const vfsTarget = this.php.readlink( vfsPath );
+		this.symlinks.set( filename, vfsTarget );
+
+		await this.mountTarget( vfsTarget, target );
+	}
+
+	/**
+	 * Mount a target path in the PHP runtime.
+	 * @param vfsTarget
+	 * @param hostTarget
+	 * @private
+	 */
+	private async mountTarget( vfsTarget: string, hostTarget: string ) {
+		let mountInfo = this.mountedTargets.get( vfsTarget );
+
+		if ( typeof mountInfo === 'undefined' ) {
+			mountInfo = {
+				preExisting: this.php.fileExists( vfsTarget ),
+				referenceCount: 0,
+				unmountFunction: null,
+			};
+			this.mountedTargets.set( vfsTarget, mountInfo );
+		}
+
+		mountInfo.referenceCount++;
+
+		if ( mountInfo.preExisting || mountInfo.unmountFunction !== null ) {
+			return;
+		}
+
+		this.php.mkdir( vfsTarget );
+		mountInfo.unmountFunction = await this.php.mount(
+			vfsTarget,
+			createNodeFsMountHandler( hostTarget )
+		);
+	}
+
+	/**
+	 * Delete a symlink from the PHP runtime.
+	 *
+	 * @param filename
+	 */
+	private async deleteMountedTarget( filename: string ) {
+		const vfsTarget = this.symlinks.get( filename );
+
+		if ( vfsTarget === undefined ) {
+			return;
+		}
+
+		// Get the mount info for the target
+		const mountInfo = this.mountedTargets.get( vfsTarget );
+		if ( typeof mountInfo === 'undefined' ) {
+			return;
+		}
+
+		mountInfo.referenceCount--;
+
+		// We can delete the symlink now as it is no longer needed.
+		this.symlinks.delete( filename );
+
+		// If the reference count is 0 and the unmount function is set, we can unmount the target
+		if ( mountInfo.referenceCount === 0 && mountInfo.unmountFunction && ! mountInfo.preExisting ) {
+			await mountInfo.unmountFunction();
+
+			// After unmounting, clear the mount target
+			if ( this.php.isFile( vfsTarget ) ) {
+				this.php.unlink( vfsTarget );
+			}
+
+			if ( this.php.isDir( vfsTarget ) ) {
+				this.php.rmdir( vfsTarget );
+			}
+
+			this.mountedTargets.delete( vfsTarget );
+		}
+	}
+
+	/**
+	 * Check if a file exists.
+	 * @param filename
+	 */
+	private async fileExists( filename: string ) {
+		try {
+			await fs.access( filename, fs.constants.F_OK );
+			return true;
+		} catch ( error ) {
+			if ( error instanceof Error && 'code' in error && error.code === 'ENOENT' ) {
+				return false;
+			}
+			// Rethrow unexpected errors
+			throw error;
+		}
+	}
```

Comment:
I've just realized that we already have this function defined in `lib/fs-utils.ts`:

https://github.com/Automattic/studio/blob/8e6816e2ba015c7013e681d63f9c359369df01eb/src/lib/fs-utils.ts#L7-L17

Reply:
Thanks! I wasn't aware but I removed it now and used the existing function.